stravalib/model.py

```diff
@@ -203,6 +285,13 @@ def __getattribute__(self, attr):
         return value
 
 
+class BoundClientEntity(BaseModel):
+    # Using Any as type here to prevent catch-22 between circular import and
+    # pydantic forward-referencing issues "resolved" by PEP-8 violations.
+    # See e.g. https://github.com/pydantic/pydantic/issues/1873
+    bound_client: Optional[Any] = None
```

Comment:
The "mixin" that allows binding a client object to entities

Reply:
what is the PEP-8 part of this? is it the circular import?

Reply:
one other question. Can you explain when you say bind a client object to entities... what is an example of this?

Reply:
> what is the PEP-8 part of this? is it the circular import?

Circular imports will prevent this from running at all. As I understood from the linked discussion, circular imports could be prevented by putting some import statements at the end of the module, which is against PEP-8.

Reply:
yes aboslutely - i understand better how the "binding" works and the objects interact. and yes that would be really weird to add imports to the bottom of a module and in conflict with pep 8! so i understand that element too now! thank you. thanks!